mozart++/core/any.hpp

```diff
@@ -0,0 +1,300 @@
+/**
+ * Mozart++ Template Library: Any
+ * Licensed under MIT License
+ * Copyright (c) 2019 Covariant Institute
+ * Website: https://covariant.cn/
+ * Github:  https://github.com/covariant-institute/
+ */
+
+#pragma once
+
+#include <mozart++/core/base.hpp>
+#include <mozart++/core/exception.hpp>
+#include <typeindex>
+
+namespace mpp {
+    class any;
+}
+
+class mpp::any final {
+public:
+    using typeid_t = std::type_index;
+    // 缓冲池大小，过大的值可能会适而其反
+    static constexpr size_t default_allocate_buffer_size = 16;
+    /*
+    * 分配器提供者，默认使用 Mozart++ 分配器，可根据需要替换为内存池
```

Comment:
ditto, a well-formed document comment looks like this:
```c++
/**
 * Hi
 */
```

**Pay attention to spaces.**